### PR TITLE
[DOC] Enhanced RDoc for String

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #count
- #delete
- #delete!
- #squeeze
- #squeeze!

Adds section "Multiple Character Selectors" to doc/character_selectors.rdoc.

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
Second commit adds #squeeze and #squeeze!, which I had not realized use character selectors.

File: string.c [hunk 1]
```diff
@@ -8252,10 +8252,10 @@ rb_str_delete(int argc, VALUE *argv, VALUE str)
 
 /*
  *  call-seq:
- *     str.squeeze!([other_str]*)   -> str or nil
+ *    squeeze!(*selectors) -> new_string or nil
```
Comment (peterzhu2118) [inline string.c:8255]: I don't think the return value is a new string. It's the `self` string right?

Comment (BurdetteLamar) [reply string.c]: Fixed.

File: string.c [hunk 2]
```diff
@@ -8158,10 +8158,11 @@ tr_find(unsigned int c, const char table[TR_TABLE_SIZE], VALUE del, VALUE nodel)
 
 /*
  *  call-seq:
- *     str.delete!([other_str]+)   -> str or nil
+ *    delete!(*selectors) -> new_string or nil
```
Comment (peterzhu2118) [inline string.c:8161]: I think this should also be fixed

```suggestion
 *    delete!(*selectors) -> self or nil
```

File: doc/character_selectors.rdoc [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
 
 - String#tr(selector, replacements): returns a new string.
-- String#tr!(selector, replacements): returns +self+.
+- String#tr!(selector, replacements): returns +self+ or +nil+.
 - String#tr_s(selector, replacements): returns a new string.
-- String#tr_s!(selector, replacements): returns +self+.
-- String#delete(*selectors): returns a new string.
-- String#delete!(*selectors): returns +self+.
+- String#tr_s!(selector, replacements): returns +self+ or +nil+.
 - String#count(*selectors): counts specified characters.
```
Comment (peterzhu2118) [inline doc/character_selectors.rdoc:12]: For consistency

```suggestion
- String#count(*selectors): returns the count of the specified characters.
```

File: doc/character_selectors.rdoc [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 - String#tr_s(selector, replacements): returns a new string.
-- String#tr_s!(selector, replacements): returns +self+.
+- String#tr_s!(selector, replacements): returns +self+ or +nil+.
+- String#count(*selectors): returns the count of the specified characters.
 - String#delete(*selectors): returns a new string.
-- String#delete!(*selectors): returns +self+.
-- String#count(*selectors): counts specified characters.
+- String#delete!(*selectors): returns +self+ or +nil+.
+- String#squeeze(*selectors): returns a new string.
+- String#squeeze!(*selectors): returns +self+ or +nil+.
```
Comment (peterzhu2118) [inline doc/character_selectors.rdoc:16]: Should these be in alphabetical order?

File: doc/character_selectors.rdoc [hunk 5]
```diff
@@ -65,3 +69,29 @@ In a character selector, these three characters get special treatment:
     "hello\r\nworld".delete("\r")   # => "hello\nworld"
     "hello\r\nworld".delete("\\r")  # => "hello\r\nwold"
     "hello\r\nworld".delete("\\\r") # => "hello\nworld"
+
+=== Multiple Character Selectors
+
+These instance methods accept multiple character selectors:
+
+- String#count(*selectors): counts specified characters.
```
Comment (peterzhu2118) [inline doc/character_selectors.rdoc:77]:
```suggestion
- String#count(*selectors): returns the count of the specified characters.
```

File: doc/character_selectors.rdoc (hunk 4 above)
Comment (BurdetteLamar) [reply doc/character_selectors.rdoc:16]: I've grouped those with argument replacements first, then those without.

**Review by peterzhu2118 — APPROVED** [review]

**Comment by cielavenir** [comment]
Just now I noticed this pull request, which positively resolves https://github.com/ruby/ruby/pull/645 . Thank you so much.

**Pull request merged**